code/03_focal_example.R

```diff
@@ -0,0 +1,90 @@
+# Filename: 03-focal_eample.R (2017-08-29)
+#
+# TO DO: Illustrate how a focal function works
+#
+# Author(s): Jannes MUENCHOW
+#
+#**********************************************************
+# CONTENTS-------------------------------------------------
+#**********************************************************
+#
+# 1. ATTACH PACKAGES AND DATA
+# 2. FOCAL EXAMPLE
+#
+#**********************************************************
+# 1 ATTACH PACKAGES AND DATA-------------------------------
+#**********************************************************
+
+# attach packages
+library(grid)
+library(gridExtra)
+library(sp)
+library(latticeExtra)
+library(lattice)
+library(sf)
+library(raster)
+
+# create data
+# raster
+r = raster(nrow = 3, ncol = 3, res = 0.5, 
+           xmn = -1.5, xmx = 1.5, ymn = -1.5, ymx = 1.5,
+           vals = 1:36)
+# in the book chapter we show subsetting, that's why we have to do so here also
+r[1, 1] = 0
+# create polygons
+# moving window
+poly_window = 
+  rbind(c(-1.5, 0), c(0,0), c(0, 1.5), c(-1.5, 1.5), c(-1.5, 0)) %>%
+  list %>%
+  st_polygon %>%
+  st_sfc %>%
+  st_sf(data.frame(id = 1), geometry = ., crs = 4326) %>%
+  as(., "Spatial")
+# target polygon
+poly_target =
+  rbind(c(-1, 0.5), c(-0.5, 0.5), c(-0.5, 1), c(-1, 1), c(-1, 0.5)) %>%
+  list %>%
+  st_polygon %>%
+  st_sfc %>%
+  st_sf(data.frame(id = 1), geometry = ., crs = 4326) %>%
+  as(., "Spatial")
+
+#**********************************************************
+# 2 FOCAL EXAMPLE------------------------------------------
```

Comment:
@Nowosad Yes, I have also looked at **visualraster** but was not entirely convinced. But I admit, I am a bit biased towards **lattice**. But I guess it doesn't matter which code we use here, lattice or ggplot2, since we don't show it in the book, and the results of both packages should be identical.